cairo/ethereum/cancun/trie.cairo

```diff
@@ -1765,6 +1822,35 @@ func _get_bytes32_preimage_for_key{poseidon_ptr: PoseidonBuiltin*}(
     return res;
 }
 
+func get_tuple_address_bytes32_preimage_for_key{poseidon_ptr: PoseidonBuiltin*}(
+    key: felt, dict_ptr_stop: DictAccess*
+) -> TupleAddressBytes32 {
+    alloc_locals;
+
+    // Get preimage data
+    let (local preimage_data: felt*) = alloc();
+    local preimage_len;
+    %{ get_preimage_for_key %}
+
+    // Verify preimage
+    if (preimage_len != 3) {
+        raise('preimage_len != 3');
+    }
+
+    let (preimage_hash) = poseidon_hash_many(preimage_len, preimage_data);
```

Comment:
if preimage_len needs to be 3, you don't need to populate it in a hint, just use 3.

Reply:
the hint is generic and assigns the primage len - as it could be used with preimages of any size (see _get_bytes_preimage_for_key), as such I will assign it in the hint

Reply:
the hint would work with `ids.preimage_len` already set to 3

this can just be

```
let preimage_len = 3;
%{ get_preimage_for_key }%
let (preimage_hash) = poseidon_hash_many(preimage_len, preimage_data);
```

Reply:
done